### PR TITLE
Added custom help command

### PR DESCRIPTION
**Pull request by H4ckerJ4cker**
This formats the default help command nicely while still keeping the functionality of being automatically updated. The command is also now used only in DMs to reduce clutter. :help [command] will however send every where as it is only a small embed and may be useful when telling someone about a command.
Screenshot 2020-10-16 at 16 14 08

### CODE REVIEW

**Review by Sh3llcod3 — COMMENTED** [review]
Does this pass Flake8 & Mypy checks?

**Review by Sh3llcod3 — APPROVED** [review]
Understandable, have a nice day?

**Review by 404dcd — APPROVED** [review]

**Comment by H4ckerJ4cker** [comment]
The command is also now used only in DMs to reduce clutter. :help [command] will however send every where as it is only a small embed and may be useful when telling someone about a command.

File: cdbot/cogs/help.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            title=cog.description,
+            description=f"Type ``{self.clean_prefix}help [command]`` for more info on a command.\n You can also type "
+                        f"``{self.clean_prefix}help [category]`` for more info on a different category."
+        )
+        for command in cog.get_commands():
+            embed.add_field(name=f"``{self.clean_prefix}{command.name}``", value=command.help)
+
+        await ctx.author.send(embed=embed)
+        if ctx.guild is not None:
+            await ctx.send(f"{ctx.author.mention} help info sent to DMs :)", delete_after=10)
```
Comment (thebeanogamer) [inline cdbot/cogs/help.py:40]:
```suggestion
            await ctx.send(f"{ctx.author.mention} help info sent to DMs", delete_after=10)
```

File: cdbot/cogs/help.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        )
+        for cog in mapping.keys():
+            if cog is not None:
+                if cog.get_commands():
+                    embed.add_field(name=cog.qualified_name,
+                                    value=f"``{self.clean_prefix}help {cog.qualified_name}``")
+
+        await ctx.author.send(embed=embed)
+        if ctx.guild is not None:
+            await ctx.send(f"{ctx.author.mention} help info sent to DMs :)", delete_after=10)
```
Comment (thebeanogamer) [inline cdbot/cogs/help.py:24]:
```suggestion
            await ctx.send(f"{ctx.author.mention} help info sent to DMs", delete_after=10)
```

File: cdbot/cogs/help.py (hunk 1 above)
Comment (H4ckerJ4cker) [reply cdbot/cogs/help.py:40]: I saw this coming

**Review by thebeanogamer — APPROVED** [review]
Eh, it's probably fine

**Review by david-cooke — CHANGES_REQUESTED** [review]
@thebeanogamer pretend I came up with some good threat to stop you merging this, I'm too tired rn

File: cdbot/cogs/help.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    async def send_bot_help(self, mapping):
+        ctx = self.context
+        embed = Embed(
+            color=Colour.blue(),
+            description=f"Type ``{self.clean_prefix}help [command]`` for more info on a command.\n""You can also type "
+                        f"``{self.clean_prefix}help [category]`` for more info on a category."
+        )
+        for cog in mapping.keys():
+            if cog is not None:
```
Comment (AmusedNetwork) [inline cdbot/cogs/help.py:17]: Could you use `if cog` instead of `if cog is not None` here?

File: cdbot/cogs/help.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                if cog.get_commands():
+                    embed.add_field(name=cog.qualified_name,
+                                    value=f"``{self.clean_prefix}help {cog.qualified_name}``")
+
+        await ctx.author.send(embed=embed)
+        if ctx.guild is not None:
```
Comment (AmusedNetwork) [inline cdbot/cogs/help.py:23]: Same here with `is not None`

File: cdbot/cogs/help.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            color=Colour.blue(),
+            title=cog.description,
+            description=f"Type ``{self.clean_prefix}help [command]`` for more info on a command.\n You can also type "
+                        f"``{self.clean_prefix}help [category]`` for more info on a different category."
+        )
+        for command in cog.get_commands():
+            embed.add_field(name=f"``{self.clean_prefix}{command.name}``", value=command.help)
+
+        await ctx.author.send(embed=embed)
+        if ctx.guild is not None:
```
Comment (AmusedNetwork) [inline cdbot/cogs/help.py:39]: Also here.

**Review by AmusedNetwork — CHANGES_REQUESTED** [review]

File: cdbot/cogs/help.py (hunk 3 above)
Comment (H4ckerJ4cker) [reply cdbot/cogs/help.py:17]: yes...yes i could have

**Pull request merged**